src/PostgREST/Error.hs

```diff
@@ -214,6 +216,51 @@ instance JSON.ToJSON ApiRequestError where
     "details" .= JSON.Null,
     "hint"    .= ("Try renaming the parameters or the function itself in the database so function overloading can be resolved" :: Text)]
 
+-- |
+-- If no relationship is found between a parent and a child, then it looks for the parent first.
+-- If the parent is not found then it does a fuzzy search to all the parents in the schema cache and
+-- gives the best match as suggestion. Otherwise, it does a fuzzy search to all the corresponding children
+-- of that parent and gives the best match as suggestion. If both are found, then no suggestion is given.
```

Comment:
Just to make sure I'm understanding this.

If no rel is found:

- Looks for parent suggestions if parent not found
- Looks for child suggestions if children not found and parent is found
- No suggestions if both are found

Is that correct?

Reply:
Yes, if both are found then there's an error with the hint, which is more complex to look for.